### PR TITLE
Organize CompositeRules

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Makes the composite rules duplicate less as per @pranavkm's request.

### CODE REVIEW

File: src/NuGetPackageVerifier/Rules/AssemblyHasReleaseConfigurationRule.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 namespace NuGetPackageVerifier.Rules
 {
-    public class AssemblyHasCorrectBuildConfigurationRule : AssemblyHasAttributeRuleBase
+    public class AssemblyHasReleaseConfigurationRule : AssemblyHasAttributeRuleBase
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/AssemblyHasReleaseConfigurationRule.cs]: `AssemblyIsBuiltInReleaseConfiguraitonRule`?

File: src/NuGetPackageVerifier/CompositeRules/DefaultCompositeRule.cs [hunk 2]
```diff
@@ -9,10 +9,8 @@ public class DefaultCompositeRule : IPackageVerifierRule
     {
         IPackageVerifierRule[] _rules = new IPackageVerifierRule[]
```
Comment (pranavkm) [inline src/NuGetPackageVerifier/CompositeRules/DefaultCompositeRule.cs]: @Eilon we'll keep the two rules around for the time being. This avoids the need to update the verifier json in all the repos at this point.

File: src/NuGetPackageVerifier/Rules/AssemblyHasReleaseConfigurationRule.cs (hunk 1 above)
Comment (Eilon) [inline src/NuGetPackageVerifier/Rules/AssemblyHasReleaseConfigurationRule.cs]: Meh. Why change this?

Comment (pranavkm) [inline src/NuGetPackageVerifier/Rules/AssemblyHasReleaseConfigurationRule.cs]: We had a hard time figuring out what the name of Release configuration rule was. Having `Release` in the name would have helped

Comment (Eilon) [inline src/NuGetPackageVerifier/Rules/AssemblyHasReleaseConfigurationRule.cs]: O I C. Ok fine rename it. I guess it doesn't show up anywhere outside of here.

**Pull request merged**